File: Aula4-parte2/Em_java/Mago.java
public class Mago extends Personagem{
    private int mana;
    public Mago(String nome, int vida, int nivel, int mana){
        super(nome, vida, nivel);
        this.mana = 50;
    }
    public int getMana(){
        return mana;
    }
    public void setMana(int mana){
        if(mana >= 0){
            this.mana = mana;
        }else{
            System.out.println("Mana não pode ser negativa.");
        }
    }
    @Override
    public void ficha(){
        System.out.println("Mago");
        super.ficha();
        System.out.println("Mana: " + this.mana);
    }
}

File: Aula4-parte2/Em_python/Mago.py
from Personagem import Personagem
class Mago(Personagem):
    def __init__(self, nome: str, vida: int, nivel: int):
        super().__init__(nome, vida, nivel)
        self._mana = 50
    @property
    def mana(self):
        return self._mana
    @mana.setter
    def mana(self, valor: int):
        if valor < 0:
            print("A mana não pode ser negativa.")
        else:
            self._mana = valor
            
    def ficha(self):
        print("Ficha do Mago:")
        super().ficha()
        print(f"Mana: {self._mana}")